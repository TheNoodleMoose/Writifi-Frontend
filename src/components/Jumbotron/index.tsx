import React from "react";
import styled from "styled-components";
import Button from "../Button";
import { ReactComponent as LogoSvg } from "../../assets/svgs/Logo.svg";
import { Link } from "react-router-dom";

const Jumbotron = () => {
  return (
    <JumboTronContainer>
      <ContentContainer>
        <LogoSvg width="200" height="200" />
        <TitleContainer>
          <TitleText>Endless stories at the tips of your fingers.</TitleText>
          <TitleText>Create and Discover now.</TitleText>
        </TitleContainer>

        <ButtonContainer>
          <Link to="/signup">
            <Button text="Start Discovering" />
          </Link>
          <Button text="Why Writifi?" />
        </ButtonContainer>
      </ContentContainer>
    </JumboTronContainer>
  );
};

export default Jumbotron;

const JumboTronContainer = styled.div`
  display: flex;
  width: 80%;
  justify-content: center;
  background-color: aquamarine;
  padding: 20px 0;
  min-width: 300px;
  @media (max-width: 768px) {
    flex-direction: column;
    width: 100%;
  }
`;

const ContentContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  margin: 0 20px;
`;

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 100%;
`;

const TitleText = styled.h1`
  font-size: 2em;
  text-align: center;
  color: #2c3e50;
  margin: 15px 0;
`;

const TitleContainer = styled.div``;
