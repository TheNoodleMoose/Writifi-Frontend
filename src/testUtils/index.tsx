import React, { ReactElement, ReactNode } from "react";

import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export const renderWithContext = (component: ReactElement) => {
  const history = createMemoryHistory();
  const wrapper = ({ children }: { children?: ReactNode }) => (
    <Router history={history}>{children}</Router>
  );

  const renderResults = render(component, { wrapper });
  return {
    ...renderResults,
    snapshot: () => {
      expect(renderResults.container).toMatchSnapshot();
    }
  };
};
