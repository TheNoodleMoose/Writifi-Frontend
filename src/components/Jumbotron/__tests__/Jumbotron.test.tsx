import React from "react";
import { renderWithContext } from "../../../testUtils";

import Jumbotron from "..";

it("renders correctly", () => {
  renderWithContext(<Jumbotron />).snapshot();
});
