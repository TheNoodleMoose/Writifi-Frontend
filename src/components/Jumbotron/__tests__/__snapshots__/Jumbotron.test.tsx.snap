// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-AxiKw bgxPGc"
  >
    <div
      class="sc-AxhCb bzzkfW"
    >
      <svg
        height="200"
        width="200"
      >
        Logo.svg
      </svg>
      <div
        class="sc-Axmtr hBVTto"
      >
        <h1
          class="sc-AxgMl fLNeVe"
        >
          Endless reading at the tips of your fingers.
        </h1>
        <h1
          class="sc-AxgMl fLNeVe"
        >
          Create and Discover now.
        </h1>
        <p
          class="sc-AxheI kLuIFW"
        >
          Writifi is a writing platform that focuses on the content you want to create and discover.
        </p>
      </div>
      <div
        class="sc-AxhUy dIrXSn"
      >
        <a
          href="/signup"
        >
          <button
            class="sc-AxjAm ekMElM"
            data-testid="Button"
          >
            <p
              class="sc-AxirZ gToTIx"
            >
              Start Discovering
            </p>
          </button>
        </a>
        <button
          class="sc-AxjAm ekMElM"
          data-testid="Button"
        >
          <p
            class="sc-AxirZ gToTIx"
          >
            Why Writifi?
          </p>
        </button>
      </div>
    </div>
  </div>
</div>
`;
